src/http/common.cr

```diff
@@ -59,6 +59,7 @@ module HTTP
       end
 
       name, value = parse_header(line)
+      break unless headers.valid_value?(value)
       headers.add(name, value)
```

Comment:
I'd suggest `headers.add(name, value) if headers.valid_value?(value)`

Reply:
This checks the validity twice. I suggest adding `add?` method that returns `false` if it's not valid, `true` otherwise.